internal/executor/executor.go

```diff
@@ -199,16 +227,19 @@ func (e *Executor) executeParamExtraction(execCtx *ExecutionContext) error {
 
 // CreateHandler creates an event handler function that can be used with the broker subscriber
 // This is a convenience method for integrating with the broker_consumer package
+//
+// Error handling strategy:
+// - All failures are logged but the message is ACKed (return nil)
+// - This prevents infinite retry loops for non-recoverable errors (e.g., 400 Bad Request, invalid data)
 func (e *Executor) CreateHandler() func(ctx context.Context, evt *event.Event) error {
 	return func(ctx context.Context, evt *event.Event) error {
 		result := e.Execute(ctx, evt)
 
+		// Log failure but ACK the message to prevent retry loops
+		// Non-recoverable errors (4xx, validation failures) should not be retried
```

Comment:
Should we be distinguishing between errors, potentially timeouts and some 500s could be retried. That or are we considering a DLQ?